src/VecSim/algorithms/svs/svs_tiered.h

```diff
@@ -0,0 +1,726 @@
+            return svs_index->deleteVectors(&label, 1);
+        }
+
+        bool label_exists = [&] {
+            std::shared_lock<std::shared_mutex> flat_lock(this->flatIndexGuard);
+            return this->frontendIndex->isLabelExists(label);
+        }();
+
+        if (label_exists) {
+            std::scoped_lock lock(this->flatIndexGuard, this->mainIndexGuard, this->journal_mutex);
+            if (this->frontendIndex->isLabelExists(label)) {
+                ret = this->frontendIndex->deleteVector(label);
+                assert(ret == 1 && "unexpected deleteVector result");
+                journal.emplace_back(label, false);
+            }
+            ret += svs_index->deleteVectors(&label, 1);
+            assert(ret < 2 && "deleteVector: vector duplication in both indices");
+        } else {
+            std::scoped_lock lock(this->mainIndexGuard);
+            ret += svs_index->deleteVectors(&label, 1);
+        }
+        return ret;
+    }
+
+    size_t indexSize() const override {
+        std::shared_lock<std::shared_mutex> flat_lock(this->flatIndexGuard);
+        std::shared_lock<std::shared_mutex> main_lock(this->mainIndexGuard);
+        return this->frontendIndex->indexSize() + this->backendIndex->indexSize();
+    }
+
+    size_t indexLabelCount() const override {
+        std::shared_lock<std::shared_mutex> flat_lock(this->flatIndexGuard);
+        std::shared_lock<std::shared_mutex> main_lock(this->mainIndexGuard);
+        return this->frontendIndex->indexLabelCount() + this->backendIndex->indexLabelCount();
+    }
+    size_t indexCapacity() const override {
+        std::shared_lock<std::shared_mutex> flat_lock(this->flatIndexGuard);
+        std::shared_lock<std::shared_mutex> main_lock(this->mainIndexGuard);
+        return this->frontendIndex->indexCapacity() + this->backendIndex->indexCapacity();
+    }
+
+    double getDistanceFrom_Unsafe(labelType label, const void *blob) const override {
+        double flat_dist = std::numeric_limits<double>::quiet_NaN();
+        {
+            std::shared_lock<decltype(this->flatIndexGuard)> lock(this->flatIndexGuard);
+            flat_dist = this->frontendIndex->getDistanceFrom_Unsafe(label, blob);
+        }
+        if (!std::isnan(flat_dist)) {
+            return flat_dist;
+        } else {
+            std::shared_lock<decltype(this->mainIndexGuard)> lock(this->mainIndexGuard);
+            return this->backendIndex->getDistanceFrom_Unsafe(label, blob);
+        }
+    }
+
+    VecSimIndexDebugInfo debugInfo() const override {
+        auto info = Base::debugInfo();
+        return info;
+    }
+
+    VecSimIndexBasicInfo basicInfo() const override {
+        VecSimIndexBasicInfo info = this->backendIndex->getBasicInfo();
+        info.blockSize = info.blockSize;
+        info.isTiered = true;
+        info.algo = VecSimAlgo_SVS;
+        return info;
+    }
+
+    VecSimDebugInfoIterator *debugInfoIterator() const override {
```

Comment:
as discussed, this doesn't include any svs-specific information, right?
If so, consider adding a TODO comment